app/config/HBaseRestRepositoryConfigLoader.scala

```diff
@@ -17,11 +17,14 @@ object HBaseRestRepositoryConfigLoader extends HBaseRestConfigLoader[HBaseRestRe
   override def load(rootConfig: Config, path: String): HBaseRestRepositoryConfig = {
     val config = rootConfig.getConfig(path)
     HBaseRestRepositoryConfig(
-      baseUrl = BaseUrl(config.getString("protocol"), config.getString("host"), config.getInt("port"), Option(config.getString("prefix"))),
+      baseUrl = BaseUrl(config.getString("protocol"), config.getString("host"), config.getInt("port"), NoneEmptyString(config.getString("prefix"))),
       namespace = config.getString("namespace"),
       username = config.getString("username"),
       password = config.getString("password"),
       timeout = config.getLong("timeout")
     )
   }
-}
\ No newline at end of file
+  // Convert to None if string is empty or null
```

Comment:
This comment is not actually correct as the function does not handle null.  If x is null then x.trim will throw a NullPointerException.  We are actually assuming that the argument cannot be null - as config.getString will throw an exception if there is no such config value, and it is not possible to define a value in application.conf that has the value null.

This highlights the danger of comments - they are often inaccurate, or become inaccurate over time (because the code is changed but the associated comments are not updated to reflect those changes).  Hence the advice to limit comments, and where a comment is helpful - focus on **why** you are doing something not **how** (if you need to explain the how, the code is not clear enough).